src/Http/Http.Abstractions/src/HttpRequest.cs

```diff
@@ -104,9 +104,9 @@ public abstract class HttpRequest
         public abstract Stream Body { get; set; }
 
         /// <summary>
-        /// Gets or sets the request body pipe <see cref="PipeReader"/>.
+        /// Gets the request body pipe <see cref="PipeReader"/>.
         /// </summary>
-        public abstract PipeReader BodyReader { get; set; }
+        public abstract PipeReader BodyReader { get; }
```

Comment:
I'm a huge fan of removing these setters. I think making replacing the entire feature the only way to replace the body PipeReader/PipeWriter, gets us off the hook for automagically replacing the body streams. After all, you'd only expect more experienced devs to try such a thing.